File: LinkedLists/StackLinkedlist.c
de <limits.h>
#include <stdio.h>
#include <stdlib.h>

struct stackNode {
    int data;
    struct stackNode* next;
};

struct stackNode* newNode(int data) {
    struct stackNode* stackNode = (struct stackNode*)malloc(sizeof(struct stackNode));
    stackNode->data = data;
    stackNode->next = NULL;
    return stackNode;
}

int isEmpty(struct stackNode* root){
    return !root;
}

void push(struct stackNode** root, int data){
    struct stackNode* node = newNode(data);
    node->next = *root;
    *root = node;
    printf("%d pushed to stack\n" , data);

}

int pop(struct stackNode** root){
    if (isEmpty(*root))
        return INT_MIN;
    struct stackNode* temp = *root;
    *root = (*root)->next;
    int popped = temp->data;
    free(temp);
    return popped;
}
int peek(struct stackNode* root)
{
    if (isEmpty(root))
        return INT_MIN;
    return root->data;
}

int main() {
    struct structNode* root = NULL;

    push(&root, 10);
    push(&root, 20);
    push(&root, 30);

    printf("%d popped from the stack \n", pop(&root));

    printf("top of stack is %d\n", peek(&root));
}

File: LinkedLists/StackLinkedlist.py
class stacknode:

    def __init__(self, data):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.root = None
    
    def isEmpty(self):
        return True if self.root == None else False
    def push(self, data):
        newnode = stacknode(data)
        newnode.next = self.root
        self.root = newnode
        print(newnode.data + " pushed to stack")
    def pop(self):
        if(self.isEmpty()):
            return float("-inf")
        temp = self.root
        self.root = self.root.next
        return temp.data
    def peek(self):
        print("the top of the stack is: " + self.root.data)
        return self.root.data
